src/uri.rs

```diff
@@ -0,0 +1,261 @@
+#![allow(dead_code)]
+
+use reqwest;
+
+use serde_derive::Deserialize;
+
+// these item structs should be namespaced into the v2 api.
+#[derive(Deserialize, Debug)]
+struct ItemAttribute {
+    attribute: String,
+    modifier: u32,
+}
+
+#[derive(Deserialize, Debug)]
+struct ItemDetails {
+    #[serde(rename = "type")]
+    c_type: String,
+    weight_class: String,
+    defense: u32,
+    // whats the type of infusions slots item?
+    //infusion_slots:
+    //infix_upgrade : HashMap<String,
+    // infix upgrade is of format { "id" : u32, "attributes" : Vec<ItemAttribute> }
+    // bit un certain how to tell serdea the values are of different types https://github.com/serde-rs/json/issues/144
+    attribute_adjustment: f64,
+    suffix_item_id: u64,
+    secondary_suffix_item_id: Option<String>,
+}
+
+#[derive(Deserialize, Debug)]
+struct Item {
+    name: String,
+    // maybe should be an option?
+    description: String,
+    #[serde(rename = "type")]
+    item_type: String,
+    id: u32,
+    level: u32,
+    rarity: String,
+    default_skin: u64,
+    game_types: Vec<String>,
+    flags: Vec<String>,
+    // not certain what the vec type should be.
+    restrictions: Vec<String>,
+    chat_link: String,
+    icon: String,
+    details: ItemDetails,
+}
+
+// should be usize?
+
+// todo: move this to some types thing?
+pub struct ItemId(pub u128);
+pub struct ItemStatId(pub u128);
+pub struct RecipeId(pub u128);
+pub struct ApiVersion(pub u8);
+pub struct ApiKey(pub String);
+pub struct AchievementId(pub u128);
+
+#[allow(non_camel_case_types)]
+pub enum EndPoint {
```

Comment:
Documentation?  What is the goal of this type.  Is it to exactly model the gw2 API URIs?  Is it to be the Requestor's request interface?  Depending on the answer I'd recommend different design changes.

Reply:
The basics is that its to capture all of the different end points of the gw2 api as well any helpeful information about each of the end points. I'm also looking to have this provide utility functions for bundling batches of requests into a single request. Such as 
item1 = items/1, item2 = items/2 then if a request wants say [item1, item2] this is distilled into a single request to the server rather than 2, then the items are returned as expected maintain a 1 to 1 return.